hilti/runtime/include/types/bytes.h

```diff
@@ -537,3 +538,14 @@ std::string to_string(const bytes::Charset& x, adl::tag /*unused*/);
 } // namespace detail::adl
 
 } // namespace hilti::rt
+
+// Disable JSON-ification of `Bytes`.
+//
+// As of nlohmann-json-0e694b4060ed55df980eaaebc2398b0ff24530d4 the JSON library misdetects the serialization for
+// `Bytes` on some platforms. We see this on platfoms not providing a C++17-compliant (e.g., in Cirrus' `no-toolchain`
+// task which uses gcc-9.3.0) where code in JSON wants to check whether `Bytes` can be converted to a
+// `std::filesystem::path`, but then runs into compiler issues.
+namespace nlohmann {
+template<>
+struct adl_serializer<hilti::rt::Bytes> {};
+} // namespace nlohmann
```

Comment:
FTR, this is a workaround for the issue seen e.g., [here](https://cirrus-ci.com/task/4623106609250304?logs=build#L32):

```
FAILED: hilti/runtime/CMakeFiles/hilti-rt-debug-objects.dir/src/library.cc.o 
/usr/bin/ccache /usr/bin/c++ -DHILTI_RT_BUILD_TYPE_DEBUG -I../3rdparty/justrx/include -I../3rdparty -Iinclude -I../hilti/runtime/include -I../3rdparty/fiber/include -I../3rdparty/fiber/deps/cxx-header-utils/include -O3 -DNDEBUG -fPIC -g -UNDEBUG -O0 -Wall -std=c++17 -MD -MT hilti/runtime/CMakeFiles/hilti-rt-debug-objects.dir/src/library.cc.o -MF hilti/runtime/CMakeFiles/hilti-rt-debug-objects.dir/src/library.cc.o.d -o hilti/runtime/CMakeFiles/hilti-rt-debug-objects.dir/src/library.cc.o -c ../hilti/runtime/src/library.cc
In file included from /usr/include/c++/9/filesystem:37,
                 from ../hilti/runtime/include/hilti/rt/3rdparty/nlohmann/json.hpp:3822,
                 from ../hilti/runtime/include/hilti/rt/json.h:13,
                 from ../hilti/runtime/src/library.cc:10:
/usr/include/c++/9/bits/fs_path.h: In instantiation of ‘struct std::filesystem::__cxx11::path::__constructible_from<hilti::rt::Bytes, void>’:
/usr/include/c++/9/type_traits:131:12:   required from ‘struct std::__and_<std::__not_<std::is_void<hilti::rt::Bytes> >, std::filesystem::__cxx11::path::__constructible_from<hilti::rt::Bytes, void> >’
/usr/include/c++/9/type_traits:136:12:   required from ‘struct std::__and_<std::__not_<std::is_same<hilti::rt::Bytes, std::filesystem::__cxx11::path> >, std::__not_<std::is_void<hilti::rt::Bytes> >, std::filesystem::__cxx11::path::__constructible_from<hilti::rt::Bytes, void> >’
/usr/include/c++/9/bits/fs_path.h:117:13:   required by substitution of ‘template<class _Tp1, class _Tp2> using _Path = typename std::enable_if<std::__and_<std::__not_<std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, std::filesystem::__cxx11::path> >, std::__not_<std::is_void<typename std::remove_pointer<_Tp>::type> >, std::filesystem::__cxx11::path::__constructible_from<_Tp1, _Tp2> >::value, std::filesystem::__cxx11::path>::type [with _Tp1 = hilti::rt::Bytes; _Tp2 = void]’
/usr/include/c++/9/bits/fs_path.h:190:7:   required by substitution of ‘template<class _Source, class _Require> std::filesystem::__cxx11::path::path(const _Source&, std::filesystem::__cxx11::path::format) [with _Source = hilti::rt::Bytes; _Require = <missing>]’
../hilti/runtime/include/hilti/rt/3rdparty/nlohmann/json.hpp:4271:26:   required by substitution of ‘template<class BasicJsonType, class T> decltype (nlohmann::detail::from_json(j, forward<T>(val))) nlohmann::detail::from_json_fn::operator()(const BasicJsonType&, T&&) const [with BasicJsonType = nlohmann::basic_json<>; T = hilti::rt::Bytes&]’
../hilti/runtime/include/hilti/rt/3rdparty/nlohmann/json.hpp:4932:38:   [ skipping 7 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
../hilti/runtime/include/hilti/rt/3rdparty/nlohmann/json.hpp:2280:7:   recursively required by substitution of ‘template<class Default, template<class ...> class Op, class ... Args> struct nlohmann::detail::detector<Default, typename nlohmann::detail::make_void<Op<Args ...> >::type, Op, Args ...> [with Default = nlohmann::detail::nonesuch; Op = nlohmann::detail::get_template_function; Args = {const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > >&, const hilti::rt::Bytes}]’
../hilti/runtime/include/hilti/rt/3rdparty/nlohmann/json.hpp:2280:7:   required by substitution of ‘template<template<class ...> class Op, class ... Args> using is_detected = typename nlohmann::detail::detector<nlohmann::detail::nonesuch, void, Op, Args ...>::value_t [with Op = nlohmann::detail::get_template_function; Args = {const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > >&, const hilti::rt::Bytes}]’
../hilti/runtime/include/hilti/rt/3rdparty/nlohmann/json.hpp:2283:8:   required from ‘struct nlohmann::detail::is_detected_lazy<nlohmann::detail::get_template_function, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > >&, const hilti::rt::Bytes>’
../hilti/runtime/include/hilti/rt/3rdparty/nlohmann/json.hpp:3493:8:   recursively required from ‘struct nlohmann::detail::conjunction<nlohmann::detail::negation<std::is_same<const hilti::rt::Bytes, nlohmann::detail::json_ref<nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > > > > >, nlohmann::detail::negation<std::is_same<const hilti::rt::Bytes, char> >, nlohmann::detail::negation<nlohmann::detail::is_basic_json<const hilti::rt::Bytes> >, nlohmann::detail::negation<std::is_same<const hilti::rt::Bytes, std::initializer_list<char> > >, nlohmann::detail::negation<std::is_same<const hilti::rt::Bytes, std::basic_string_view<char, std::char_traits<char> > > >, nlohmann::detail::is_detected_lazy<nlohmann::detail::get_template_function, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > >&, const hilti::rt::Bytes> >’
../hilti/runtime/include/hilti/rt/3rdparty/nlohmann/json.hpp:3493:8:   required from ‘struct nlohmann::detail::conjunction<nlohmann::detail::negation<std::is_pointer<const hilti::rt::Bytes> >, nlohmann::detail::negation<std::is_same<const hilti::rt::Bytes, nlohmann::detail::json_ref<nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > > > > >, nlohmann::detail::negation<std::is_same<const hilti::rt::Bytes, char> >, nlohmann::detail::negation<nlohmann::detail::is_basic_json<const hilti::rt::Bytes> >, nlohmann::detail::negation<std::is_same<const hilti::rt::Bytes, std::initializer_list<char> > >, nlohmann::detail::negation<std::is_same<const hilti::rt::Bytes, std::basic_string_view<char, std::char_traits<char> > > >, nlohmann::detail::is_detected_lazy<nlohmann::detail::get_template_function, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > >&, const hilti::rt::Bytes> >’
../hilti/runtime/include/hilti/rt/3rdparty/nlohmann/json.hpp:19046:73:   required by substitution of ‘template<class ValueType, typename std::enable_if<nlohmann::detail::conjunction<nlohmann::detail::negation<std::is_pointer<_Tp> >, nlohmann::detail::negation<std::is_same<ValueType, nlohmann::detail::json_ref<nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > > > > >, nlohmann::detail::negation<std::is_same<ValueType, char> >, nlohmann::detail::negation<nlohmann::detail::is_basic_json<BasicJsonType> >, nlohmann::detail::negation<std::is_same<ValueType, std::initializer_list<char> > >, nlohmann::detail::negation<std::is_same<ValueType, std::basic_string_view<char, std::char_traits<char> > > >, nlohmann::detail::is_detected_lazy<nlohmann::detail::get_template_function, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > >&, ValueType> >::value, int>::type <anonymous> > nlohmann::basic_json<>::operator ValueType<ValueType, <enumerator> >() const [with ValueType = const hilti::rt::Bytes; typename std::enable_if<nlohmann::detail::conjunction<nlohmann::detail::negation<std::is_pointer<_Tp> >, nlohmann::detail::negation<std::is_same<ValueType, nlohmann::detail::json_ref<nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > > > > >, nlohmann::detail::negation<std::is_same<ValueType, char> >, nlohmann::detail::negation<nlohmann::detail::is_basic_json<BasicJsonType> >, nlohmann::detail::negation<std::is_same<ValueType, std::initializer_list<char> > >, nlohmann::detail::negation<std::is_same<ValueType, std::basic_string_view<char, std::char_traits<char> > > >, nlohmann::detail::is_detected_lazy<nlohmann::detail::get_template_function, const nlohmann::basic_json<std::map, std::vector, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, long int, long unsigned int, double, std::allocator, nlohmann::adl_serializer, std::vector<unsigned char, std::allocator<unsigned char> > >&, ValueType> >::value, int>::type <anonymous> = <missing>]’
../hilti/runtime/src/library.cc:20:13:   required from here
/usr/include/c++/9/bits/fs_path.h:113:31: error: ‘std::__cxx11::basic_string<char>’ is an inaccessible base of ‘hilti::rt::Bytes’
  113 |       : decltype(__is_path_src(std::declval<_Source>(), 0))
      |                  ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```